src/ipc.rs

```diff
@@ -62,12 +62,12 @@ pub fn bytes_channel() -> Result<(IpcBytesSender, IpcBytesReceiver),Error> {
 }
 
 #[derive(Debug)]
-pub struct IpcReceiver<T> where T: Deserialize + Serialize {
+pub struct IpcReceiver<T> where T: for<'de> Deserialize<'de> + Serialize {
```

Comment:
@dtolnay: Ah yep, it was DeserializeOwned that I was thinking of.

@Eijebong: It would clean up all of the type signatures.  That's about it though.